### PR TITLE
Enable nullability in some DataGridView.Methods members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some DataGridView.Methods members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/10020)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.Methods.cs [hunk 1]
```diff
@@ -5909,7 +5912,7 @@ private void DrawSplitBar(Rectangle r)
     private void EditingControls_CommonMouseEventHandler(object sender, MouseEventArgs e, DataGridViewMouseEvent dgvme)
     {
         Debug.Assert(_ptCurrentCell.X != -1);
-        int adjustedX = _editingPanel.Location.X + e.X;
+        int adjustedX = _editingPanel!.Location.X + e.X;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.Methods.cs:5915]: add one more assert and remove bang `!`?

**Review by dreddy-work — APPROVED** [review]
LGTM.

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.Methods.cs (hunk 1 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.Methods.cs:5915]: @dreddy-work I did that yesterday.

**Review by dreddy-work — APPROVED** [review]

**Comment by gpetrou** [comment]
@dreddy-work looks like one of the builds randomly failed.

**Pull request merged**